Add/fix links and fix typos

mlvandijk opened the pull request:
#Partial fix for https://github.com/cucumber/docs.cucumber.io/issues/19

Add / fix all the links I could find a place to link to. Just leaves the following: 
* implementations/jvm.md:
[[Android Runner]] - 4x
[[Data Table]] - 2x
* gherkin-reference.md
[[Multiline Step Arguments]] - 2x
* cucumber/running-features.md
[[IronRuby]]
* cucumber/state.md
[[glue path]]
[[Configuration Options]]
* cucumber/step-definitions
[[Multiline Step Arguments]] 
* ruby/calling-steps-from-stepdefinitions
[[Multiline Step Arguments]]
* ruby/ruby-on-rails [Spork and --drb]] 
* wiki/autotest-integration 
[[Troubleshooting]]

Suggestions to fix remaining:
1. Add info on Android runner (but not by me)
2. Add file on Multiline Step Arguments, including Data Tables (unless this was intentionally left out / removed?)
3. Remove links to IronRuby, glue path, Configuration Options, Spork and --drd, Troubleshooting (unless someone can tell me what they should point to)

mlvandijk committed:
Add link to Related tools in blog-post

mlvandijk committed:
Add .idea to .gitignore

mlvandijk committed:
Fix link in continuous-integration.md

mlvandijk committed:
Fix links in cucumber.yml.md

mlvandijk committed:
Fix typo in feature-coupled-step-definition-antipattern.md

mlvandijk committed:
Fix typos in hooks.md

mlvandijk committed:
Partial fix links in running-features.md

mlvandijk committed:
Partial fix links in state.md

mlvandijk committed:
Partial fix links in step-definitions.md

mlvandijk committed:
Fix links in tags.md

mlvandijk committed:
Fix links in background.md

mlvandijk committed:
Fix typo in conjunction-steps-antipattern.md

mlvandijk committed:
Partial fix links in ruby-on-rails.md

mlvandijk committed:
Fix typo in perl.md

mlvandijk committed:
Add link in getting-started-with-cucumber-jvm

mlvandijk committed:
Add links in cucumber-backgrounder.md

mlvandijk committed:
Add links in introduction_to_cucumber_and_bdd.md

olleolleolle commented on content/cucumber/continuous-integration.md:
```diff
@@ -43,8 +43,7 @@ If you're using Ant, you can run cucumber with the [`exec`](https://ant.apache.o
 
-Follow the install instructions here:
-<https://github.com/jenkinsci/cucumber-reports-plugin>
+Follow the install instructions [here](https://github.com/jenkinsci/cucumber-reports-plugin).
```
Can you include the name of the plugin in the link text? "Here" is a little context-free

mlvandijk replied:
Done

olleolleolle commented on content/implementations/ruby/ruby-on-rails.md:
```diff
@@ -173,11 +173,12 @@ html: --format html --out features.html
 ### Special tags
 
-There are two special [[tags]] you can use to change how Cucumber runs your scenarios
+There are two special [tags](/cucumber/tags) you can use to change how Cucumber runs your scenarios
```
Can this sentence end with a colon?

mlvandijk replied:
Done

mxygem commented:
- I didn't know this but IronRuby is a .NET Ruby release like JRuby for Java. Link: http://ironruby.net/ 
- I'm not sure how crucial this is right now (it's certainly out of scope for this link fixing PR anyway) but I noticed the tags.md file uses the old tag syntax of `--tags ~@important` which will throw warnings in 3.x

Past that, LGTM! Nice catch on some of the grammar stuff outside the links.

olleolleolle reviewed (commented):
I made tiny comments!

mlvandijk commented:
@js Thanks! Added a link to ironruby, leaving the tags for now
@olleolleolle Thanks! 
Will process comments hopefully this evening

mxygem commented:
@mlvandijk - `not @tag` basically. Here's more: https://github.com/cucumber/cucumber/tree/master/tag-expressions

mlvandijk committed:
Fix review comments and add/fix more links

mlvandijk commented:
Also added a link to Spork and removed tags for 'glue path' 'Cucumber options' and 'Troubleshooting'.

This leaves 'Data Tables' / 'Multiline step arguments' which I think could be covered by adding the part about StepArguments (DocSring/DataTabels) as it is here: https://cucumber.io/docs/reference
And it leaves 'Android Runner'.

mlvandijk commented:
Sorry @js - missed you comment about tags. Will update later.
